src/main/java/net/openhft/chronicle/network/FatalTcpHandlerException.java

```diff
@@ -0,0 +1,11 @@
+package net.openhft.chronicle.network;
+
+/**
+ * Thrown when a TCP handler encounters an unrecoverable error
+ */
+public class FatalTcpHandlerException extends RuntimeException {
```

Comment:
maybe should descend from InvalidEventHandlerException?

Reply:
I like the idea, but becayse InvalidHandlerException is unchecked it means adding `throws InvalidEventHandlerException` to a lot of places, some of which it doesn't make that much sense (e.g. `NetworkContextManager.nc(...)`).

WireTcpHandler calls `onInitialize` in the nc setter. Perhaps this is an indication that onInitialize should be a separate explicit call?
https://github.com/OpenHFT/Chronicle-Network/blob/068999a2e06655933e2683659407a5d77f15776e/src/main/java/net/openhft/chronicle/network/WireTcpHandler.java#L339-L343